src/Analyzer.Core/TemplateAnalyzer.cs

```diff
@@ -97,9 +100,48 @@ public IEnumerable<IEvaluation> AnalyzeTemplate(string template, string paramete
                     throw new TemplateAnalyzerException(BicepCompileErrorMessage, e);
                 }
             }
+            IEnumerable<IEvaluation>  evaluations = DeepAnalyzeTemplate(template, parameters, templateFilePath, template, 0, isBicep, sourceMap);
+
+            // For each rule we don't want to report the same line more than once
+            // This is a temporal fix
+            var evalsToValidate = new List<IEvaluation>();
+            var evalsToNotValidate = new List<IEvaluation>();
+            foreach (var eval in evaluations)
+            {
+                if (!eval.Passed && eval.Result != null)
+                {
+                    evalsToValidate.Add(eval);
+                }
+                else
+                {
+                    evalsToNotValidate.Add(eval);
+                }
+            }
+            var uniqueResults = new Dictionary<(string, int), IEvaluation>();
+            foreach (var eval in evalsToValidate)
+            {
+                uniqueResults.TryAdd((eval.RuleId, eval.Result.LineNumber), eval);
+            }
+            evaluations = uniqueResults.Values.Concat(evalsToNotValidate);
+
+            return evaluations;
+        }
 
+        /// <summary>
+        /// Runs TemplateAnalyzer logic covering nested templates
```

Comment:
```suggestion
        /// Analyzes ARM templates, recursively going through the nested templates
```

Or something like that mentions the recursion at least